src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs

```diff
@@ -0,0 +1,726 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using Microsoft.Azure.Management.PolicyInsights;
+using Microsoft.Azure.Management.PolicyInsights.Models;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using PolicyInsights.Tests.Helpers;
+using Xunit;
+
+namespace PolicyInsights.Tests
+{
+    public class PolicyInsightsTests : TestBase
+    {
+        // Change this to Record when you want to record a test; also change the project to build for full .net since interactive login won't work for .net core
+        private static readonly HttpRecorderMode Mode = HttpRecorderMode.Playback;
+
+        #region Test setup
+
+        private static readonly string SubscriptionId = "d0610b27-9663-4c05-89f8-5b4be01e86a5";
+
+        static PolicyInsightsTests()
+        {
+            if (HttpRecorderMode.Record == Mode)
+            {
+                Environment.SetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=" + SubscriptionId + ";Environment=Prod;HttpRecorderMode=Record;");
+                Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Record");
+            }
+            else
+            {
+                Environment.SetEnvironmentVariable("TEST_CSM_ORGID_AUTHENTICATION", "SubscriptionId=" + SubscriptionId + ";Environment=Prod;HttpRecorderMode=Playback;");
```

Comment:
@bulentelmaci you never want to set anything in Playback because you are not authenticating anything. All you are doing is replaying your recorded files.
Also you don't want to set anything in code during record mode. What you should be doing is, save the connection string for your recording in a keyVault. Please take away all your dependencies hard coded

Reply:
Without the subscription ID, playback mode failed for me, hence I did this. I can remove all these hardcoded stuff for sure, but it will make life painful again next time we want to record sessions (will need an app id, and using interactive login wont work, etc.).